planner/core/rule_partition_processor.go

```diff
@@ -978,6 +978,9 @@ func (p *rangePruner) extractDataForPrune(sctx sessionctx.Context, expr expressi
 		// If the partition expression is col, use constExpr.
 		constExpr = con
 	}
+	if !constExpr.ConstItem(sctx.GetSessionVars().StmtCtx) {
```

Comment:
Please add a comment here to include the cases in the issue and inform that `constExpr` may not really be a constant when coming here.

Reply:
added